sysutils/pfSense-pkg-apcupsd/files/usr/local/www/widgets/widgets/apcupsd.widget.php

```diff
@@ -0,0 +1,407 @@
+
+			$ph = popen("/usr/local/sbin/apcaccess -h " . escapeshellarg($nisip) . ":" . escapeshellarg($nisport) . " 2>&1", "r" );
+			while ($v = fgets($ph)) {
+				$results[trim(explode(': ',$v)[0])]=trim(explode(': ',$v)[1]);
+			}
+			pclose($ph);
+
+			$rtnstr = "";
+
+			$rtnstr .= "<tr><td>Status</td><td colspan=\"3\">\n";
+			
+			if($results != null) {
+				switch ($results['STATUS']) {
+					default:
+					case 'ONBATT':
+						if (str_replace(" Percent", "", $results['BCHARGE']) <= 25) {
+							$rtnstr .= "<span class=\"fa fa-battery-empty\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 50) {
+							$rtnstr .= "<span class=\"fa fa-battery-quarter\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 75) {
+							$rtnstr .= "<span class=\"fa fa-battery-half\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 99) {
+							$rtnstr .= "<span class=\"fa fa-battery-three-quarters\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						} else {
+							$rtnstr .= "<span class=\"fa fa-battery-full\" style=\"color:red;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:red;font-style:bold;font-size:2em\">\n";
+						}
+						break;
+					case 'CHARGING':
+						if (str_replace(" Percent", "", $results['BCHARGE']) <= 25) {
+							$rtnstr .= "<span class=\"fa fa-battery-empty\" style=\"color:orange;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:orange;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 50) {
+							$rtnstr .= "<span class=\"fa fa-battery-quarter\" style=\"color:orange;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:orange;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 75) {
+							$rtnstr .= "<span class=\"fa fa-battery-half\" style=\"color:orange;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:orange;font-style:bold;font-size:2em\">\n";
+						} else if (str_replace(" Percent", "", $results['BCHARGE']) <= 99) {
+							$rtnstr .= "<span class=\"fa fa-battery-three-quarters\" style=\"color:orange;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:orange;font-style:bold;font-size:2em\">\n";
+						} else {
+							$rtnstr .= "<span class=\"fa fa-battery-full\" style=\"color:orange;font-size:2em;transform: rotate(270deg);\"/><span style=\"color:orange;font-style:bold;font-size:2em\">\n";
+						}
+						break;
```

Comment:
IF I read it right, the difference between this block and above is color being orange when it's charging.  This also can be combined with previous block just setting a `$color` variable.  It will simplify a lot the code here